app/parse.py

```diff
@@ -8,9 +19,145 @@
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_all_pages_quotes() -> [Quote]:
+    # logging
+    time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    logging.info(f"{time} - Start parsing quotes")
+
+    page_text = requests.get(BASE_URL).content
+    page_soup = BeautifulSoup(page_text, "html.parser")
+
+    all_quotes = get_single_page_quotes(page_soup)
+
+    # next button searching
+    next_element = page_soup.select_one(".next")
+    while next_element is not None:
+        # Checking the existence of an element and the href attribute
+        if next_element and next_element.a and "href" in next_element.a.attrs:
+            next_href = next_element.a["href"]
+            # Using a Regular Expression to Extract a Number from a URL
+            match = re.search(r"/page/(\d+)/", next_href)
+            if match:
+                next_page_number = match.group(1)
+
+                # logging
+                time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+                logging.info(
+                    f"{time} - Start parsing page #{next_page_number}"
+                )
+
+                next_page_url = urljoin(
+                    BASE_URL,
+                    f"/page/{next_page_number}/"
+                )
+                next_page_text = requests.get(next_page_url).content
+                next_page_soup = BeautifulSoup(
+                    next_page_text,
+                    "html.parser"
+                )
+
+                # Adding quotes from a new page
+                all_quotes.extend(get_single_page_quotes(next_page_soup))
+
+                # next button searching
+                next_element = next_page_soup.select_one(".next")
+
+    return all_quotes
+
+
+def write_quotes_to_csv(output_csv_path: str) -> None:
+    quotes = get_all_pages_quotes()
+    with open(output_csv_path, "w", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(QUOTE_FIELDS)
+        writer.writerows([astuple(quote) for quote in quotes])
+
+
+def format_name(input_name: str) -> str:
+    # Removing diacritics
+    normalized_name = unicodedata.normalize("NFD", input_name)
+    name_without_accents = "".join(
+        char for char in normalized_name if unicodedata.category(char) != "Mn"
+    )
+    # Replace spaces and dots with hyphens
+    formatted_name = re.sub(
+        r"[.\s]+", "-",
+        name_without_accents
+    ).replace("'", "").rstrip("-")
+    return formatted_name
+
+
+def get_all_authors_biography(authors: list) -> [Author]:
+    returning_authors_list = []
+    for name in authors:
+        refactored_name = format_name(name)
+        author_url = urljoin(BASE_URL, f"/author/{refactored_name}/")
+        page_text = requests.get(author_url).content
+        first_page_soup = BeautifulSoup(page_text, "html.parser")
+        biography = first_page_soup.select_one(
+            ".author-description"
+        ).text.strip()
```

Comment:
The code assumes that the `.author-description` element is always present. This could lead to errors if the element is missing. Please add error handling to manage cases where elements are not found.